// flkit/lib/util/token.js
'use strict';

exports.__esModule = true;
exports.SourceLocation = undefined;

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * SourceLocation
 */

var SourceLocation =
/**
 * constructor
 * @param  {} p []
 * @return {}   []
 */
exports.SourceLocation = function SourceLocation() {
  var p = arguments.length <= 0 || arguments[0] === undefined ? {} : arguments[0];
  var start = arguments[1];
  var end = arguments[2];
  (0, _classCallCheck3.default)(this, SourceLocation);

  this.start = start;
  this.end = end;
  this.source = '';
  if (p.sourceFile !== null) {
    this.source = p.sourceFile;
  }
};

/**
 * token
 */


var Token =
/**
 * constructor
 * @param  {} p []
 * @return {}   []
 */
function Token() {
  var p = arguments.length <= 0 || arguments[0] === undefined ? {} : arguments[0];
  (0, _classCallCheck3.default)(this, Token);

  this.type = p.type;
  this.value = p.value;
  this.start = p.start;
  this.end = p.end;
  this.loc = new SourceLocation(p, p.startLoc, p.endLoc);
  this.range = [p.start, p.end];
};

exports.default = Token;
